### ISSUE TITLE
Color picker

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#1807]
# Color picker

## Brief description
A color picker allows users to select a color from a predefined list or custom color palette, and create custom colors.

Color pickers help users immediately style visual elements such as categorization tags, data visualizations, and white-labeling without abandoning the user interface.  Products can choose the level of customization that they need based on their individual use cases.

## Details

[Link to Design](https://pages.github.ibm.com/ai-applications/design/components/color-picker/usage/)
This is also being added to the [PAL site](https://github.ibm.com/CDAI-design/pal/issues/841).

### Links to other materials
[Box assets](https://ibm.ent.box.com/folder/158680830247?s=2ytg7knwi6b75nlmb6sn5xthhdgil3xn)

## Design Maintainer(s)
- Diana Tran

## Tasks

Before starting work on this epic, please review and complete the following.

- [ ] Initial review of design/existing code

### Working in Carbon for IBM Products package

- [ ] Have you reviewed our
      [contribution guidelines](https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/master/.github/CONTRIBUTING.md)?
- [ ] Have you noted our
      [code guidelines](https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/master/docs/CODE_GUIDELINES.md)?
- [ ] Did you use the CLI to scaffold your new component?

**Comment by Laura-Marshall** [comment]
Note:
For dark theme in the Figma kit implementation, we realised that changing the colour selector to black had poor contrast. As not documented, we decided to hard code this white and keep it the same across Gray 10 and 100. Once implemented, we can change the tokens to match the implementation. 

### The approach we went with
![image (5)](https://github.com/carbon-design-system/ibm-products/assets/53219208/553a1f0c-cb26-49cf-ad2c-53ad4b28875c)

![image (6)](https://github.com/carbon-design-system/ibm-products/assets/53219208/4642e5ff-46c1-4525-b77e-a7d43a5a178c)

### The approach we didn't go with
![image (7)](https://github.com/carbon-design-system/ibm-products/assets/53219208/e6f8e97b-0c2c-4cf2-85c9-73f8125e61f1)

**Comment by elycheea** [comment]
In our case, we sometimes use the hardcoded color tokens. We might be able to look at localized theming as well although I wonder if the hardcoded approach is more appropriate here since it’s only the selector and not the whole area that needs to be localized. 🤔 

### Alternate view

White selector on g10 | White selector on g100 | Grey selector on g100*
----------------------|-------------------------|-----------------------
![image (5)](https://github.com/carbon-design-system/ibm-products/assets/53219208/553a1f0c-cb26-49cf-ad2c-53ad4b28875c) | ![image (6)](https://github.com/carbon-design-system/ibm-products/assets/53219208/4642e5ff-46c1-4525-b77e-a7d43a5a178c) | ![image (7)](https://github.com/carbon-design-system/ibm-products/assets/53219208/e6f8e97b-0c2c-4cf2-85c9-73f8125e61f1)

* not recommended due to lack of contrast

**Comment by elycheea** [comment]
@sstrubberg Let’s close this as not planned and open style picker (but in the icebox) instead? We had another team reach out requesting style picker so might be good to start getting a clearer picture of that demand in its own issue.
